lib/ui/painting.dart

```diff
@@ -4936,6 +4936,59 @@ enum ClipOp {
   intersect,
 }
 
+/// Defines a list of glyphs to draw, used by [Canvas.drawGlyphRun].
+///
+/// All glyphs are baseline aligned when drawing.
+class GlyphRun {
+
+  /// Creates a glyph run.
+  ///
+  /// The [glyphs] and [positions] lengths must match.
+  GlyphRun({
+    required this.glyphs,
+    required List<Offset> positions,
+    required this.fontFamily,
+    required this.fontSize,
+    this.fontWeight = FontWeight.normal,
+    this.fontStyle = FontStyle.normal,
+  })  : assert(glyphs.length == positions.length),
+        positions = _encodePointList(positions);
+
+  /// Creates a glyph run from raw data.
+  ///
+  /// The length of [positions] must be twice the length of [glyphs].
+  GlyphRun.raw({
+    required this.glyphs,
+    required this.positions,
+    required this.fontFamily,
+    required this.fontSize,
+    this.fontWeight = FontWeight.normal,
+    this.fontStyle = FontStyle.normal,
+  }) : assert(positions.length == glyphs.length * 2);
+
+  /// The id of the glyphs to draw. If a glyph id in this list does not exist
+  /// in the font given by [familyName], it takes no visual effect.
```

Comment:
Here and below: style guide says that the first line of a doc comment should be a single-line sentence. 

We should document here how to get a glyph ID, what a glyph ID is, and importantly that it may be different for different fonts or even different versions of a font.